Framework/Indexing/inc/MantidIndexing/DetectorID.h

```diff
@@ -34,16 +33,13 @@ namespace Indexing {
   File change history is stored at: <https://github.com/mantidproject/mantid>
   Code Documentation is available at: <http://doxygen.mantidproject.org>
 */
-template <class T1, class T2,
-          class = typename std::enable_if<std::is_integral<T1>::value &&
-                                          std::is_integral<T2>::value>::type>
-std::vector<T2> makeRange(T1 first, T2 last) {
-  std::vector<T2> vec(last - static_cast<T2>(first) + 1);
-  std::iota(vec.begin(), vec.end(), static_cast<T2>(first));
-  return vec;
-}
+struct MANTID_INDEXING_DLL DetectorID
```

Comment:
That's a shame.

However I think having proper types for the different index types is conceptually much better and clearer. I would hope this change doesn't affect performance but I think that would be worth checking.